pvlib/atmosphere.py

```diff
@@ -337,6 +337,82 @@ def gueymard94_pw(temp_air, relative_humidity):
     return pw
 
 
+def rh_from_tdew(temperature, dewpoint, coeff=(6.112, 17.62, 243.12)):
+    """
+    Calculate relative humidity from dewpoint temperature using the Magnus equation.
+
+    Parameters
+    ----------
+    temperature : numeric
+        Air temperature (dry-bulb temperature) in degrees Celsius
+    dewpoint : numeric
+        Dewpoint temperature in degrees Celsius
+    coeff: tuple
+        Magnus equation coefficient (A, B, C)
+
+    Returns
+    -------
+    numeric
+        Relative humidity as percentage (0.0-100.0)
+
+    References
+    ----------
+    .. [1] "Guide to Instruments and Methods of Observation",
+       World Meteorological Organization, WMO-No. 8, 2023.
+       https://library.wmo.int/idurl/4/68695
+    """
+
+    # Calculate vapor pressure (e) and saturation vapor pressure (es)
+    e = coeff[0] * np.exp((coeff[1] * temperature) / (coeff[2] + temperature))
+    es = coeff[0] * np.exp((coeff[1] * dewpoint) / (coeff[2] + dewpoint))
+
+    # Calculate relative humidity as percentage
+    relative_humidity = 100 * (es / e)
+
+    return relative_humidity
+
+
+def tdew_from_rh(
+    temperature, relative_humidity, coeff=(6.112, 17.62, 243.12)
+):
+    """
+    Calculate dewpoint temperature using Magnus equation.
+    This is a reversal of the calculation in :py:func:`rh_from_tdew`.
+
+    Parameters
+    ----------
+    temperature : numeric
+        Air temperature (dry-bulb temperature) in degrees Celsius
+    relative_humidity : numeric
+        Relative humidity as percentage (0-100)
+
+    Returns
+    -------
+    numeric
+        Dewpoint temperature in degrees Celsius
+
+    References
+    ----------
+    .. [1] "Guide to Instruments and Methods of Observation",
+       World Meteorological Organization, WMO-No. 8, 2023.
+       https://library.wmo.int/idurl/4/68695
+    """
+    # Calculate the term inside the log
+    # From RH = 100 * (es/e), we get es = (RH/100) * e
+    # Substituting the Magnus equation and solving for dewpoint
+
+    # First calculate ln(es/A)
+    ln_term = (
+        (coeff[1] * temperature) / (coeff[2] + temperature)
+        + np.log(relative_humidity/100)
```

Comment:
looks like RH zero might throw an error or warning here

Reply:
Should we add a check for zero values or leave the original message in the stack trace?

Reply:
I don't think it's necessary, RH=0% is practically impossible ([source](https://www.city-data.com/forum/weather/1273770-just-how-low-can-relative-humidity.html)).